doc/use-on-icub-head.md

```diff
@@ -0,0 +1,62 @@
+use the robotology-superbuild on the computer contained in the head of the iCub robot. 
+
+**Warning: if your iCub robot is not currently using the robotology-superbuild, do not attempt to migrate it
+without coordinating with the iCub support.  For any doubt, please open an issue at https://github.com/robotology/icub-tech-support.**
+
+## Prepare your system 
+The operating system contained in the `icub-head`/`pc104` is tipically installed by IIT, for more information on it, please
+check the relevant documentation at http://wiki.icub.org/wiki/The_Linux_on_the_pc104 . On this machine, 
+in `/home/icub/.bashrc_iCub` a script containing several enviroment variables definitions is provided. If you want to  migrate to use
+the robotology-superbuild, you need to remove the existing `.bashrc_iCub` file, and substitute it with the one provided in https://git.robotology.eu/MBrunettini/icub-environment/raw/master/bashrc_iCub_superbuild.
+After you modified the `.bashrc_iCub` script, reboot the computer and  in a new shell check che values of the `ROBOTOLOGY_SUPERBUILD_SOURCE_DIR` and `YARP_ROBOT_NAME` env variables.
+If `ROBOTOLOGY_SUPERBUILD_SOURCE_DIR` contains `/usr/local/src/robot/robotology-superbuild`, and `YARP_ROBOT_NAME` contains the string specific to your robot,
+then the modification of the `.bashrc_iCub` was successful. 
+
+## Get software source code, compile and install
+On the `icub-head`/`pc104`, the software repository necessary to run the iCub robot are contained in the `/usr/local/src/robot` directory.
+On a new machine, this directory should be empty. All the relevant software can download and build with the following commands:
+~~~sh
+cd /usr/local/src/robot
+git clone https://github.com/robotology/robotology-superbuild.git
+cd robotology-superbuild
+mkdir build
+cmake -DROBOTOLOGY_USES_GAZEBO:BOOL=OFF -DROBOTOLOGY_ENABLE_ICUB_HEAD:BOOL=ON ..
+make
+~~~
+The main difference over the standard installation of the robotology-superbuild, is that the `ROBOTOLOGY_USES_GAZEBO` option
+is disabled (as the Gazebo is tipically not installed in the `icub-head`/`pc104` machine) and the `ROBOTOLOGY_ENABLE_ICUB_HEAD` option
+is enabled, to enable all the YARP devices that are necessary to interface with the internal communication bus of the iCub robot.
+The `make` command in this case will download, compile and install all the software necessary to run the robot itself.
+
+If you need to use other *superbuild profiles*, they can be easily enabled with the corresponding CMake option.
+
+**Important: do not run `make install` for the superbuild: the superbuild already installs all the software in 
+`/usr/local/src/robot/robotology-superbuild/build/install`, and installing the software elsewhere on the robot is not currently supported.**
+
+**Important: If you are using an old iCub that still uses the CAN internal bus, you also need to enable the `ROBOTOLOGY_USES_CFW2CAN` CMake options. 
```

Comment:
Related to `CWF2CAN`:

the current `bashrc_iCub` on `iCubGenova01` has some additional lines, i.e.

```
# For compiling
DRIVER_ROOT=/lib/modules/$(uname -r)/iCubDrivers/
# to be used if the corresponding hardware is mounted on the robot
#export ESDCANAPI_DIR=${DRIVER_ROOT}/esdCanApi/
#export PLXCANAPI_DIR=${DRIVER_ROOT}/plxCanApi/
export CFW2CANAPI_DIR=${DRIVER_ROOT}/cfw002/
```

with respect to the `bashrc_iCub` specified in this README.

Since I copied it from the old `iCubGenova01` pc104 USB stick but I was not present during the first `robotology-superbuild` setup on the `iCubGenova01`, I don't know whether this is required or not.

Reply:
The definition of `CFW2CANAPI_DIR` is indeed strictly necessary to to enable the `ROBOTOLOGY_USES_CFW2CAN` and compile  successfully the software, but unfortunately it is not present in the https://git.robotology.eu/MBrunettini/icub-environment/blob/master/bashrc_iCub_superbuild file . Even for that, we need to open a ticket. I can handle this.

Reply:
Ok, I will open the other one for the networking issue then.

Reply:
> The definition of `CFW2CANAPI_DIR` is indeed strictly necessary to to enable the `ROBOTOLOGY_USES_CFW2CAN` and compile successfully the software, but unfortunately it is not present in the https://git.robotology.eu/MBrunettini/icub-environment/blob/master/bashrc_iCub_superbuild file . ~Even for that, we need to open a ticket. I can handle this.~ I reported the issue on the IIT internal tracker.

In the meanwhile I also added a clarification phrase in https://github.com/robotology/robotology-superbuild/pull/212/commits/510b643e0ea1568dbc5447c59d11c99c50f8ce6d .